packages/menu/src/Menu.ts

```diff
@@ -119,18 +116,26 @@ export class Menu extends SpectrumElement {
     }
 
     public focusMenuItemByOffset(offset: number): void {
+        const step = offset || 1;
         const focusedItem = this.menuItems[this.focusedItemIndex] as MenuItem;
```

Comment:
If `this.focusedItemIndex` is the last menu item, and I programmatically remove that item, could the logic here lead to issues?

The thing I'm struggling with is that `focusedItemIndex` can be a valid index, but point to something that's not focused. While that doesn't cause issues with the code as-is (except maybe this), it feels fragile.

Reply:
This is a great question! 

The `childList` or `subtree` of the `sp-menu` element are currently being observed for mutations at which time the following method is called:
```
public updateSelectedItemIndex(): void {
        let index = this.menuItems.length - 1;
        let item = this.menuItems[index] as MenuItem;
        while (index && item && !item.selected) {
            index -= 1;
            item = this.menuItems[index] as MenuItem;
        }
        index = Math.max(index, 0);
        this.menuItems.forEach((item, i) => {
            if (i !== index) {
                item.focused = false;
            }
        });
        this.focusedItemIndex = index;
        this.focusInItemIndex = index;
    }
```
This steps through all of the updated items and reassigns the value of `this.focusedItemIndex` to either the index of an item with `selected` or the zeroth item in the list. That means almost all cases `this.focusedItemIndex` will point to am existing item. I'm seeing this case:
```
selectedItem.remove(); // remove the focused item
menu.focus(); // immediately focus the menu
// in the case that it was the last item and no replacement item is applied nothing will be visual selected.
```
I'll address this and add a test to track regressions.

Reply:
@adixon-adobe I think we're coving this use case in a good way now. Let me know what you think!